kontrol-service/plugins/plugins.go

```diff
@@ -37,58 +37,66 @@ func NewPluginRunner(gitPluginProvider GitPluginProvider, tenantId string, db *d
 	}
 }
 
-func (pr *PluginRunner) CreateFlow(pluginUrl string, serviceSpec corev1.ServiceSpec, deploymentSpec appv1.DeploymentSpec, flowUuid string, arguments map[string]string) (appv1.DeploymentSpec, string, error) {
+func (pr *PluginRunner) CreateFlow(pluginUrl string, serviceSpec corev1.ServiceSpec, originalWorkloadSpec kardinal.WorkloadSpec, flowUuid string, arguments map[string]string) (kardinal.WorkloadSpec, string, error) {
+	workloadSpec := originalWorkloadSpec.DeepCopy()
+
 	repoPath, err := pr.getOrCloneRepo(pluginUrl)
 	if err != nil {
-		return appv1.DeploymentSpec{}, "", fmt.Errorf("failed to get or clone repository: %v", err)
+		return workloadSpec, "", fmt.Errorf("failed to get or clone repository: %v", err)
 	}
 
 	serviceSpecJSON, err := json.Marshal(serviceSpec)
 	if err != nil {
-		return appv1.DeploymentSpec{}, "", fmt.Errorf("failed to marshal service spec: %v", err)
+		return workloadSpec, "", fmt.Errorf("failed to marshal service spec: %v", err)
 	}
 
-	deploymentSpecJSON, err := json.Marshal(deploymentSpec)
+	deploymentSpecJSON, err := json.Marshal(workloadSpec.GetTemplateSpec())
 	if err != nil {
-		return appv1.DeploymentSpec{}, "", fmt.Errorf("failed to marshal deployment spec: %v", err)
+		return workloadSpec, "", fmt.Errorf("failed to marshal deployment spec: %v", err)
 	}
 
 	result, err := runPythonCreateFlow(repoPath, string(serviceSpecJSON), string(deploymentSpecJSON), flowUuid, arguments)
 	if err != nil {
-		return appv1.DeploymentSpec{}, "", err
+		return workloadSpec, "", err
 	}
 
 	var resultMap map[string]json.RawMessage
 	err = json.Unmarshal([]byte(result), &resultMap)
 	if err != nil {
-		return appv1.DeploymentSpec{}, "", fmt.Errorf("failed to parse result: %v", err)
+		return workloadSpec, "", fmt.Errorf("failed to parse result: %v", err)
 	}
 
-	var newDeploymentSpec appv1.DeploymentSpec
-	err = json.Unmarshal(resultMap["deployment_spec"], &newDeploymentSpec)
-	if err != nil {
-		return appv1.DeploymentSpec{}, "", fmt.Errorf("failed to unmarshal deployment spec: %v", err)
+	if resultMap["pod_spec"] == nil {
+		logrus.Warnf("No pod_spec found in plugin result")
```

Comment:
I wonder if we should eject here? because for instance the RDS plugin could be executed and if the dev service doesn't overwrite the env vars for the DB connection the new DEV service will connect to the production DB instead, affecting it.

Reply:
Fair point! Better be safe than sorry